text/0000-router-link-and-helpers.md

```diff
@@ -0,0 +1,161 @@
+- add named arguments to `LinkTo`: `@routeName` and `@models`
+- add built-in helpers for custom link-like behavior
+
+*Phase 2*
+- add `Link` component with `@to` named argument -- all other named arguments will be the params
+- deprecate `LinkTo`
+
+*Phase 3*
+- remove `LinkTo` and the corresponding deprecation
+
+
+#### Phase 1
+*Add Named Arguments to `LinkTo`*
+
+  `@routeName` and `@models` will allow us to smoothly transition away from `{{link-to}}` into the `<AngleBracket />` invocation-style.
+
+  - example:
+
+  ```hbs
+    {{!-- before --}}
+    {{#link-to 'posts.edit' post.id}}
+      Edit
+    {{/link-to}}
+
+    {{!-- After --}}
+    <LinkTo @routeName='posts.edit' @models={{hash postId=post.id}}>
+      Edit
+    </LinkTo>
+  ```
+
+
+  The `<AngleBracket />` invocation with named arguments is more verbose, but trade-off here is the added clarity of what argument is used for what purpose -- especially with respect to the models / model ids.
+
+*Helpers to be implemented*
+
+ - `transition`
+    this is a helper that would tie into the router to be able to achieve the same functionality as the navigational usage of `LinkTo`
+
+    examples:
+
+      ```hbs
+      <button {{action (transition to='posts.edit' postId=post.id)}} />
+        Edit
+      </button>
+
+      <button {{action (transition to='login' replace=true)}} />
+        Login
+      </button>
+      ```
+
+
+
+ - `is-route-active`
+    this returns a boolean representing whether or not the current route matches the passed argument.
+
+    example:
+
+      ```hbs
+      <button
+        {{action (transition to='posts.edit' post=model.post)}}
+        class={{if (is-route-active 'posts.edit' post=model.post) 'selected'}} />
+      ```
+
+   this may need to support multiple invocation styles, for when there aren't parameters and the dev wants to be concise:
+
+   examples:
+
+     ```hbs
+     <button ... class={{if (is-route-active 'posts') 'selected'}} />
+      No model params provided
+     </button>
+     ```
+
+#### Phase 2
+
+*Add `Link` Component*
+
+`Link` could just be tagless component with the following template:
+
+```hbs
+{{!-- components/link/template.hbs --}}
+<a
+  {{action (transition @to models=@models)}}
+  class="{{if (is-route-active @to models=@models) 'active'}}">
+
+  {{yield}}
+
+</a>
+```
+
+Usage:
+```hbs
+<Link @to='posts.edit' @models={{post}}>Edit Post</Link>
+```
+
+where `@models` can be a `hash`, `array`, or just a single object or id.
+
+
+
+
+**Deprecation** `LinkTo` aka `link-to`
+
+The goal of `Link` and the route helpers is to provide a flexible way of routing from the template while providing a sample component with sensible defaults. This would achieve the exact same functionality as `LinkTo`, so `LinkTo` would no longer be needed and could eventually be removed.
+
+It's possible we could write a codemod to auto-convert everyone's non-angle-bracket invocation of `{{#link-to ...` to the new angle bracket component: `Link`
```

Comment:
This feels more of an open question than a detailed design, due to the uncertainty of the proposal.